### ISSUE TITLE
primitives outside of scissor region are not removed from the input list

### ISSUE THREAD

**Issue by vacarsu** [boydm/scenic#255]
Primitives with the input style can still be interacted with even if they are outside of a scissor region.

**Comment by crertel** [comment]
My wager (@boydm correct me on this) is that we don't account for stenciling when handling input events--if you look at [`Scenic.Input.t`](https://github.com/boydm/scenic/blob/414104923f780e838a4e87ceb078569d2c6945da/lib/scenic/view_port/input.ex#L76-L87) it looks like input events don't mention where they come in from.

My first guess as to how to handle this is to pass up the visible component where an input hit happens along with the normal information, maybe using stencil buffer picking or something.

**Comment by boydm** [comment]
Yeah. it does not account for stenciling when checking for inputs.

**Comment by boydm** [comment]
This is going to require some thought

**Comment by boydm** [comment]
And also yes, [#253](https://github.com/boydm/scenic/issues/253) would be the same issue.

**Comment by boydm** [comment]
My inital thought is to mark scissor regions in the input tree in the viewport. Then when searching down the tree, it would be pretty easy to see if the point is in the scissor region or not. If yes, continue searching. If not, abandon this branch of the tree.

**Comment by vacarsu** [comment]
Yeah, in most cases you don't want to account for stenciling. I remember that being a big pain point in Scenic 0.10.0 where any render primitive would block input for any primitive underneath it. It made things like scroll containers impossible.
